pkg/azurefile/controllerserver.go

```diff
@@ -90,8 +90,8 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 			fileShareName = v
 		case diskNameField:
 			diskName = v
-		case fsTypeField:
-			fsType = v
+		case vhdFsTypeField:
```

Comment:
we may still need to support the original fstype parameter if backwards compatibility is important. Will let @andyzhangx decide.

Reply:
Yeah, I think we need to get rid of them before it goes to 1.0, but yeah... let's see

Reply:
I think we still need to make backwards compatibility, is it possible to keep this `fsType`, e.g. we have two places to set this field: 1) `csi.k8s.io/fstype`  2) `fstype: ext4`  
```
if `csi.k8s.io/fstype` != "" {
  fsType = `csi.k8s.io/fstype` setting
} esle {
  fsType = `fstype` setting
}
```

Azure disk driver also has same issue, it's better not rename or remove existing parameters.

Reply:
Hmm, so I think at least one thing important is that we should read fsType from VolumeCapability instead of parameters.

And I feel like use fsType to decide when to use VHD over normal Azure file can be confusing. Would it possible to add a field vhdFsType and put that as the recommended field to set? And we can support fsType at the same time. But remove it once the driver wants to promote GA

Reply:
And another reference:

Here is how gce-pd-csi-driver parses the fsType: 
https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/2f5ea47b054d091c9c97362f2e48721a7f8de698/pkg/gce-pd-csi-driver/node.go#L306

Reply:
looks like we have 3 parameters now:
 - `parametes.fsType`
 - `parametes.vhdFsType`
 - `csi.k8s.io/fstype`

if we want to use `parametes.vhdFsType`, then had better not use `csi.k8s.io/fstype` since the naming is different, another option is only use:
 - `parametes.fsType`
 - `csi.k8s.io/fstype`

It's still open to discuss. Thanks.

Reply:
Unfortunately because of https://github.com/kubernetes-csi/external-provisioner/issues/328, any PV's fstype is going to default to ext4 even if nothing was specified in the StorageClass.

That means existing PVs are going to have fstype ext4 by default, even if they were using cifs. So I think to retain backwards compatibility, we cannot use 'csi.k8s.io/fstype' to determine if something should be vhd or not. We need a specific parameter for vhd which then determines how we interpret the 'csi.k8s.io/fstype' field.

Reply:
Hmm I think my point is that regardless how fsType are passing from parameters. The fsType should only be decided by VolumeCapabilities. And that is the way moving forward for CSI spec compatibility. Currently Azuredisk and Azurefile does not seem to read from VolumeCapabilities to get fsType.

Reply:
it's better that only add `vhdFsType` in this PR, and keep `fsType` as compatible, and due to this issue(https://github.com/kubernetes-csi/external-provisioner/issues/328), `csi.k8s.io/fstype` could be useless now since it's always `ext4` by default.

Reply:
The external provisioner issue has been fixed in 2.0. However existing PVs are going to have PV.fstype set to ext4